packages/stacking/src/index.ts

```diff
@@ -0,0 +1,410 @@
+    burnBlockHeight: number;
+  }) {
+    const { hashMode, data } = convertBTCAddress(poxAddress);
+    const hashModeBuffer = bufferCV(new BN(hashMode, 10).toBuffer());
+    const hashbytes = bufferCV(data);
+    const address = tupleCV({
+      hashbytes,
+      version: hashModeBuffer,
+    });
+    const [contractAddress, contractName] = contract.split('.');
+    const network = this.network; 
+    const txOptions: ContractCallOptions = {
+      contractAddress,
+      contractName,
+      functionName: 'stack-stx',
+      // sum of uStx, address, burn_block_height, num_cycles
+      functionArgs: [
+        uintCV(amountMicroStx.toString(10)),
+        address,
+        uintCV(burnBlockHeight),
+        uintCV(cycles),
+      ],
+      validateWithAbi: true,
+      network,
+    };
+    return txOptions;
+  }
+
+  /**
+   * Check stacking status
+   * 
+   * @returns {Promise<StackerInfo>} that resolves to a StackerInfo object if the operation succeeds
+   */
+  async getStatus(): Promise<StackerInfo> {
+    const [contractAddress, contractName] = (await this.getPoxInfo()).contract_id.split('.');
+    const functionName = 'get-stacker-info';
+
+    return callReadOnlyFunction({
+      contractAddress,
+      contractName,
+      functionName,
+      senderAddress: this.address,
+      functionArgs: [
+        standardPrincipalCV(this.address)
+      ],
+      network: this.network
+    })
+    .then((responseCV: ClarityValue) => {
+      if (responseCV.type === ClarityType.OptionalSome) {
+        const someCV = responseCV as SomeCV;
+        const tupleCV: TupleCV = someCV.value as TupleCV;
+        const poxAddress: TupleCV = tupleCV.data["pox-addr"] as TupleCV;
+        const amountMicroStx: UIntCV = tupleCV.data["amount-ustx"] as UIntCV;
+        const firstRewardCycle: UIntCV = tupleCV.data["first-reward-cycle"] as UIntCV;
+        const lockPeriod: UIntCV = tupleCV.data["lock-period"] as UIntCV;
+        const version: BufferCV = poxAddress.data['version'] as BufferCV;
+        const hashbytes: BufferCV = poxAddress.data['hashbytes'] as BufferCV;
+      
+        return {
+          amountMicroStx: amountMicroStx.value.toString(),
+          firstRewardCycle: firstRewardCycle.value.toNumber(),
+          lockPeriod: lockPeriod.value.toNumber(),
+          poxAddress: {
+            version: version.buffer,
+            hashbytes: hashbytes.buffer
+          },
+          btcAddress: '',
+        }
+      } else if (responseCV.type === ClarityType.OptionalNone) {
+        throw new Error(`No stacking status found for ${this.address}`);
+      } else {
+        throw new Error(`Error fetching stacker info`);
+      }
+    });
+  }
+
+  /**
+   * Adjust microstacks amount for locking after taking into account transaction fees
+   * 
+   * @returns {StacksTransaction} that resolves to a transaction object if the operation succeeds
+   */
+  modifyLockTxFee({ tx, amountMicroStx }: { tx: StacksTransaction; amountMicroStx: BN }) {
```

Comment:
can this be `setStackingTxFee`?

Reply:
That name seems confusing since it doesn't actually set the fee, but modifies the locked amount.

Reply:
right, makes sense